apps/designer/app/designer/features/workspace/canvas-tools/text-toolbar/text-toolbar.tsx

```diff
@@ -49,16 +64,57 @@ const onClickPreventDefault: MouseEventHandler<HTMLDivElement> = (event) => {
 
 type ToolbarProps = {
   css?: CSS;
-  onValueChange: (value: Value) => void;
   rootRef: React.Ref<HTMLDivElement>;
+  textToolbar: TextToolbarValue;
+  publish: Publish;
 };
 
-const Toolbar = ({ css, onValueChange, rootRef }: ToolbarProps) => {
+const Toolbar = ({ css, rootRef, textToolbar, publish }: ToolbarProps) => {
```

Comment:
I used onValueChange here to decouple this render component from publish function.
Trying to keep publish higher up in the tree so that majority of leaf components are built without application-level knowledge.